lib/src/layer/polygon_layer.dart

```diff
@@ -27,6 +38,10 @@ class Polygon {
   final TextStyle labelStyle;
   final PolygonLabelPlacement labelPlacement;
   final bool rotateLabel;
+  // Designates whether the given polygon points follow a clock or anti-clockwise direction.
+  // This is respected during draw call batching for filled polygons. Otherwise, batched polygons
+  // of opposing clock-directions cut holes into each other leading to a leaky optimization.
+  final bool filledAndClockwise;
```

Comment:
This is extra responsibility for the user to get right. Can you give a little more detail, or perhaps think of a way around needing this?

Reply:
I'm not sure that it adds responsibility to the user considering that the value is set by the [constructor](https://github.com/fleaflet/flutter_map/pull/1599/files#diff-963bb1a45c292ea8c2c6558adae2bf75441edeb0dec681bf4235dac4b2bcf026R67) but maybe it should be in a private variable 🤔

Reply:
Ah, sorry missed that. Agreed, in that case it should be private.

Reply:
Making it private sounds good... done. Note that this change is explicitly intended to require less responsibility from users, i.e. you don't have to get your normals right to not be screwed by the batching details